test/functional/test_framework/test_framework.py

```diff
@@ -194,7 +194,10 @@ def parse_args(self):
                             help="Run test using legacy wallets", dest='descriptors')
 
         self.add_options(parser)
-        self.options = parser.parse_args()
+        # Running TestShell in a Jupyter notebook causes additional args to be passed
+        # (e.g -f notebook-file.json)
+        # Using parse_known_args ensures we only parse the arguments defined above
+        self.options, _ = parser.parse_known_args()
```

Comment:
I like the error message on typos of the arguments and slightly prefer the alternative solution presented here: https://stackoverflow.com/a/56349168/2084795

Reply:
Nice idea. Hopefully it doesn't end up like this: https://stackoverflow.com/a/67280166 (I don't know though any details about Jupyter though, on which situations which extra arguments are passed etc.)

Reply:
great call out, @MarcoFalke , i confirmed this does silently accept typos in cli args, which would be super hard to debug when using the test args from the command line.

i updated to the linked solution and, per @theStack 's comment, confirmed this was the only extra argument being passed. it doesn't guarantee new arguments wont pop up in the future and break this, but it seems better to use this approach for now and if/when more arguments break it, we can reevaluate the whole approach for a more robust solution.